server/src/QRServer/db/connector.py

```diff
@@ -387,6 +397,77 @@ async def _generate_ranking(self, start_date: datetime, end_date: datetime) -> L
             ))
         return ranking_entries
 
+    async def get_user_rating(self, user_id: str, month: int, year: int) -> Optional[UserRating]:
+        c = await self.conn.cursor()
+        await c.execute(
+            "select"
+            " rating,"
+            " revision"
+            " from user_ratings"
+            " where user_id = ?"
+            " and month = ?"
+            " and year = ?",
+            (user_id, month, year)
+            )
+        row = await c.fetchone()
+        if row:
+            return UserRating(user_id, month, year, rating=row[0], revision=row[1])
+        return None
+
+    async def update_users_rating(self, winner_id: str, loser_id: str, month: int, year: int):
+        # CONSTS
+        RETRIES = 3
+        for _ in range(RETRIES):
```

Comment:
Retry logic should not be mixed with logic that is being retried (SRP)

Reply:
fixed